Add support for Twenty Twenty-One theme

jonathansadowski opened the pull request:
### All Submissions:

* [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

<!-- Mark completed items with an [x] -->

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

This PR adds support for the Twenty Twenty-One theme.

Closes #28134 

### How to test the changes in this Pull Request:

1. Install the Twenty Twenty-One theme by following the [instructions on the Github Repo](https://github.com/WordPress/twentytwentyone#installation).
2. Move the theme from `twentytwentyone-trunk` to `twentytwentyone` and make sure it is still active or activate it after doing this. This is necessary because of the checks in `theme_support_includes`.
3. Apply this PR 

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [ ] Have you successfully run tests with your changes locally?

<!-- Mark completed items with an [x] -->

### Changelog entry

> Enhancement: Add compatibility for Twenty Twenty-One theme.


Issues fixed, with screenshots:

Fixes #28276 
![image](https://user-images.githubusercontent.com/363749/99274031-692a9f00-27ef-11eb-8ce6-d3c01b4d43f9.png)

Fixes #28274 
![image](https://user-images.githubusercontent.com/363749/99274241-b3138500-27ef-11eb-9b38-bd76a40d530d.png)

Fixes #28273 
![image](https://user-images.githubusercontent.com/363749/99274371-e0603300-27ef-11eb-999a-56540f6838bc.png)

Fixes #28272
![image](https://user-images.githubusercontent.com/363749/99274498-05ed3c80-27f0-11eb-8b55-39990d211b02.png)

jeffikus committed:
Adds structure for Twenty Twenty One default theme support. Issue #27766
- New class class-wc-twenty-twenty-one.php
- New stylesheet twenty-twenty-one.scss
- Updates checks for default themes in theme_support_includes() and wc_is_wp_default_theme_active()

jonathansadowski committed:
Use twentytwenty as base

jonathansadowski committed:
Clean up 'onsale' styling

jonathansadowski committed:
Use currentColor for body-color to support contrasting background colors

jonathansadowski committed:
Improve product title styling

jonathansadowski committed:
Update content area padding, adjust padding around empty notices

jonathansadowski committed:
Update checkout page max-width

jonathansadowski committed:
Remove button color overrides; #site-content to #main

roykho commented:
So Twenty Twenty One is seeing this issue as well https://github.com/woocommerce/woocommerce/issues/27656 I've fixed it in two other themes in this PR here https://github.com/woocommerce/woocommerce/pull/27997

Commit:
Fix product image stretching #28274

Commit:
Fix related products

Commit:
Fix additional info spacing #28272

Commit:
Darkmode support on checkout payment box

Commit:
Display block overlay on checkout

Commit:
Apply related products fix to all products

Commit:
Fix stretched product admin images

Commit:
Fix save payment method checkbox in checkout

Commit:
Adjust margin on price, related product h2, and variations table

Commit:
Fix related products alignment at lower breakpoints

Commit:
Improve the readability of cart confirmation

Commit:
Improve margin beneath short description

Commit:
Improve content-area padding at smaller viewports

rodrigoprimo committed:
Fix the layout of banner in the home page for the 2021 theme

rodrigoprimo committed:
Adjust font-weight of the section titles in the home page for the 2021 theme

Konamiman committed:
Fix overlapping elements in shop items.

rodrigoprimo committed:
Start adjusting the product categories widget in the home page for the 2021 theme

Commit:
2021 Compat: Update the style of the "sale" badge

Commit:
Update styling for homepage category

Commit:
Update badge to have dynamic width

vedanshujain committed:
Style fixes for cart page for 2021 theme.

vedanshujain committed:
More style fixes

jonathansadowski committed:
Merge pull request #28371 from woocommerce/update/2021-onsale-badge

Twenty Twenty-One Compat: Adjust Sale Badge

Commit:
Calculate width for product li; improve smaller viewport

Commit:
Remove non-WooCommerce specific wp-block-cover styles

jonathansadowski committed:
Merge branch 'add/2021-compat' into update/twenty-twenty-one-home-page-css

Commit:
Remove non-WooCommerce specific h2 styles

ObliviousHarmony committed:
Added appropriate styling to checkout page

ObliviousHarmony committed:
Fixed the apply_coupon button

vedanshujain committed:
Merge fixups

Konamiman committed:
Fixes in the "My Account" page for 2021 theme

- Adjust padding under the "My Account" title (was too big)
- Adjust color of menu items (was too light)
- Adjust thickness of underline on menu items hover (was too thick)
- Add some space between the "Hello" line and the next paragraph

vedanshujain committed:
Removed old styles that conflicts with new.

Konamiman committed:
Changes on "My Account" menu items based on mockup from designer:

- Even darker
- Smaller font
- Reduce space between items
- Un-bold

ObliviousHarmony committed:
Adjusted the checkbox visibility

ObliviousHarmony committed:
Adjusted the checkbox

ObliviousHarmony committed:
Adjusted the dark mode

jonathansadowski committed:
Merge pull request #28381 from woocommerce/add/2021-compat-checkout-page

Add Checkout Page compatibility to Twenty Twenty-One

jonathansadowski committed:
Merge branch 'add/2021-compat' into fix/28304

Commit:
Remove cart padding; set proceed to checkout button width to 100%

jonathansadowski committed:
Merge pull request #28387 from woocommerce/fix/28304

Make cart page visually compatible with 2021 theme

jonathansadowski commented:
Carrying over some issues remaining from https://github.com/woocommerce/woocommerce/pull/28387#pullrequestreview-537673358 found by @roykho 

> So these vertical lines should be removed ![](https://d.pr/i/lnZtDR+)

> The apply coupon button should have no fill ![](https://d.pr/i/TgHcmm+)

> The update cart button should have no fill ![](https://d.pr/i/jdlJnV+)

> The cart totals area should have no border lines ![](https://d.pr/i/1XQVWj+)

ObliviousHarmony committed:
Merge pull request #28343 from woocommerce/update/twenty-twenty-one-home-page-css

Add home page compatibility for Twenty Twenty-One theme

Commit:
Fix height of add to cart button

ObliviousHarmony committed:
Corrected the "on sale" badge display

ObliviousHarmony committed:
Corrected cart page according to design

Commit:
Make tabs look tabby

ObliviousHarmony committed:
Removed bullet points from shipping method selection during checkout

Commit:
Adjust the margin around display of variable product price on single product page

ObliviousHarmony committed:
Merge pull request #28333 from woocommerce/add/2021-compat-single-product

Add single product page compatibility for Twenty Twenty-One

jonathansadowski committed:
Merge pull request #28395 from woocommerce/update/2021-compat-cart

Twenty Twenty-One: Cart Page

jonathansadowski committed:
Merge branch 'add/2021-compat' into add/2021-compat-my-account-page

Commit:
Update targeting for account page styles

jonathansadowski committed:
Merge pull request #28385 from woocommerce/add/2021-compat-my-account-page

Add "My Account" page compatibility for Twenty Twenty-One

Commit:
Update the my account page colors

ObliviousHarmony reviewed (approved):
LGTM!

This first pass will do for 4.8.0-RC.1 and we can resolve all of the outstanding issues. Thanks for all of the great work everyone!
